ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/AdminResponseHandlers.java

```diff
@@ -47,18 +156,22 @@ static void handleListStreamsResponse(
           .collect(Collectors.toList())
       );
     } catch (Exception e) {
-      cf.completeExceptionally(new IllegalStateException(
-          "Unexpected server response format. Response: " + streamsListEntity));
+      return Optional.empty();
```

Comment:
Do we really want to swallow the exception here? It might make sense to propagate the exception to the completable future in the calling method.

Reply:
Do you mean the specific exception `e` or the newly created `IllegalStateException`? The illegal state exception is still being sent to the CompletableFuture, it just happens in `handleListStreamsResponse()` now rather than here. As for the specific exception `e`, I guess there could be value in seeing the error message. What's your recommendation for exposing the error message from this method? Should we create a composite return type to contain either a `List<StreamInfo>` or an error message / exception? Feels rather heavy-weight though I can't think of anything better.

Reply:
Merging this PR for now to unblock the PRs behind it. Can add this in as a follow-on. 

@purplefox do you have recommendations for how to propagate any errors encountered from these methods to the calling methods? I can only think of either creating a composite return type (to contain either the successful response or an error / error message) or having the calling method optionally pass in a consumer to receive an error message, but both feel like overkill for getting an error message back from a method. There has to be a simpler way...?